cdhweb/pages/migrations/0002_homepage.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.27 on 2020-01-08 19:59
+from __future__ import unicode_literals
+
+from django.db import migrations
+from cdhweb.pages.migration_utils import add_child, create_revision
+
+
+def create_homepage(apps, schema_editor):
+    '''Create a new wagtail HomePage with any existing content from an old
+    Mezzanine home page, and delete Wagtail's default welcome page. Point the
+    default Wagtail site at the new HomePage.'''
+
+    Page = apps.get_model('wagtailcore', 'Page')
+    Site = apps.get_model('wagtailcore', 'Site')
+    HomePage = apps.get_model('cdhpages', 'HomePage')
+
+    # check for an existing mezzanine 'home' page and save its content
+    try:
+        RichTextPage = apps.get_model('pages', 'RichTextPage')
+        old_home = RichTextPage.objects.get(title='Home')
+        content = old_home.content
+    except (LookupError, RichTextPage.DoesNotExist):
+        content = ''
+
+    # create the new homepage underneath site root
+    root = Page.objects.get(title='Root')
+    home = add_child(apps, root, HomePage, title='Home')
+
+    # create a new revision for the page to apply content & mark migration
+    create_revision(apps, home, content=content)
+
+    # point the default site at the new homepage
+    site = Site.objects.first()
+    site.root_page = home
+    site.root_page_id = home.id
+    site.save()
+
+    # delete the default welcome page
+    welcome = Page.objects.get(title='Welcome to your new Wagtail site!')
+    welcome.delete()
+
+
+def revert_homepage(apps, schema_editor):
```

Comment:
Revert logic has to be provided for the `TestMigration` class to work, since it has to know how to rollback the migration it applies in the test. This is also why `streamfield_to_html` exists